python/bind_data_access.h

```diff
@@ -120,6 +120,15 @@ class DataAccessHelper {
   };
 };
 
+inline void expect_scalar(const Dimensions &dims, const std::string_view name) {
+  if (dims != Dimensions{}) {
+    std::ostringstream oss;
+    oss << "The '" << name << "' property cannot be used with non-scalar "
+        << "Variables. Got dimensions " << to_string(dims);
```

Comment:
Minor: It seems for many of the cases where you use `expect_scalar` you would also be able to hint what would be appropriate. For example value -> values, variance -> variances? Might give a more helpful message as we assume users?

Reply:
:+1:

But I just noticed that `.variance` does not complain if the variable has no variance, it simply returns `None`. Do you think this is the correct behaviour or should it throw an exception when you use the singular property erroneously?

Reply:
Yes I think there is a difference between (1) having no variance because unset. And (2) logically not having singular variance because has dimension.